spec/Section 6 -- Execution.md

```diff
@@ -513,9 +622,14 @@ CollectFields(objectType, selectionSet, variableValues, visitedFragments):
     - Append {selection} to the {groupForResponseKey}.
   - If {selection} is a {FragmentSpread}:
     - Let {fragmentSpreadName} be the name of {selection}.
-    - If {fragmentSpreadName} is in {visitedFragments}, continue with the next
-      {selection} in {selectionSet}.
-    - Add {fragmentSpreadName} to {visitedFragments}.
+    - If {fragmentSpreadName} provides the directive `@defer` and its {if}
+      argument is {true} or is a variable in {variableValues} with the value
+      {true}:
+      - Let {deferDirective} be that directive.
```

Comment:
This currently implies that `... @defer { __typename }` will result in the selection set _NOT_ being deferred since it does not have a `true` `if` argument (and this argument has no default).

Reply:
I rewrote this, inverting the logic, which is closer to what is used for for the include directive further up in this function.

>  If {fragmentSpreadName} provides the directive `@defer` and its {if}
      argument is not {false} or is a variable in {variableValues} with the
      value {false}:

That should make it clear that only `if: false` will prevent the fragment from being deferred